src/nagini_translation/translators/call.py

```diff
@@ -149,6 +149,35 @@ def _translate_super(self, node: ast.Call, ctx: Context) -> StmtsAndExpr:
         else:
             raise InvalidProgramException(node, 'invalid.super.call')
 
+    def translate_adt_cons(self, cons: PythonClass, args: List, pos,
+                           ctx: Context) -> Expr:
+        """
+        Constructs ADTs via a sequence of constructor calls and
+        boxing/unboxing calls.
+        """
+        info = self.no_info(ctx)
+        adt_name = cons.adt_def.name + '_ADT'
+        adt_prefix = cons.adt_def.name + '_'
+        adt_type = self.viper.DomainType(adt_name, {}, [])
+
+        # If expected argument type is the ADT type (another constructor call),
+        # unbox translated argument
+        for index, ((_, arg_type), translated_arg) in enumerate(zip(cons.fields.items(),
```

Comment:
If you only want the values, not the names, you can use fields.values() instead of fields.items()

Reply:
Good, updated.